appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -31,22 +31,20 @@
  * with the cloud logging agent.
  *
  */
-public class VmRuntimeFileLogHandler extends FileHandler {
-
+public class VmRuntimeFileLogHandler extends FileHandler implements SystemLogger {
   // This exists for testing purposes only.  If set, the cloud logger may lose logs.
   public static final String LOG_DIRECTORY_PROPERTY = "com.google.apphosting.logs";
   public static final String LOG_PATTERN_CONFIG_PROPERTY =
       "com.google.apphosting.vmruntime.VmRuntimeFileLogHandler.pattern";
   // Log files to /var/log/app_engine/app.[0-2].log.json
   private static final String DEFAULT_LOG_DIRECTORY = "/var/log/app_engine";
-  private static final String DEFAULT_LOG_PATTERN = "app.%g.log.json";
+  private static final String DEFAULT_LOG_PATTERN = "app.%u.%g.log.json";
   private static final String APP_ENGINE_LOG_CONFIG_PATTERN_ENV = "APP_ENGINE_LOG_CONFIG_PATTERN";
-  private static final int LOG_MAX_SIZE = 100 * 1024 * 1024;
+  private static final int LOG_PER_FILE_SIZE = 100 * 1024 * 1024;
```

Comment:
Status is, not implemented.

The following properties are supported now (this is default behavior of java.util.logging):

 *  `com.google.apphosting.vmruntime.VmRuntimeFileLogHandler.limit` is the `LOG_PER_FILE_SIZE`
 * `com.google.apphosting.vmruntime.VmRuntimeFileLogHandler.count` is the `LOG_MAX_FILES`

But the problem is, you cannot default those values to anything meaningful, they HAVE to be defined for those values to be used (otherwise they result in no-value, or default behavior).  You cannot reset those values after the construction of the Handler too, they either have to be passed into the Handler constructor *or* be pulled from the properties.  There's no other way.

It also makes no sense for this specific VmRuntimeFileLogHandler to allow property based configuration, as this isn't configured from the WebApp side, its a System Logger that doesn't use the properties files to initialize.

This logger is initialized from ServiceLoader, at startup of Jetty, with some options configurable via system properties and runtime environment variables.

Also, the webapp side logging has no access to even initialize this VmRuntimeFileLogHandler (due to the ClassLoader isolation) anyway, meaning you couldn't set this up and configure it as part of the webapp side logging.properties file.

The only meaningful way to configure this is outside of the webapp, at the server / system level.
Do we want to support that?   I would discourage that, as you would then have hierarchies of property files (which is often confusing for people).

Example scenario:

 * Cloud logging.properties - initialized VmRuntimeFileLogHandler (so that fluent can access the logs)
 * Jetty's logging.properties - initializes the jetty's slf4j-api -> slf4j-jdk14 layer, and adds the RequestLoggerHandler (this file could also have jetty specific logging levels and possibly extra handlers)
 * Compat level, webapp logging.properties - configures the webapp's specific logging requirements.